src/qt/bitcoin.cpp

```diff
@@ -259,9 +259,26 @@ void BitcoinApplication::createPaymentServer()
 }
 #endif
 
-void BitcoinApplication::createOptionsModel(bool resetSettings)
+bool BitcoinApplication::createOptionsModel(bool resetSettings)
 {
-    optionsModel = new OptionsModel(node(), this, resetSettings);
+    optionsModel = new OptionsModel(node(), this);
+    if (resetSettings) {
+        optionsModel->Reset();
+    }
+    bilingual_str error;
+    if (!optionsModel->Init(error)) {
+        fs::path settings_path;
+        if (gArgs.GetSettingsPath(&settings_path)) {
+            error += Untranslated("\n");
+            std::string quoted_path = strprintf("%s", fs::quoted(fs::PathToString(settings_path)));
+            error.original += strprintf("Settings file %s might be corrupt or invalid.", quoted_path);
+            error.translated += tr("Settings file %1 might be corrupt or invalid.").arg(QString::fromStdString(quoted_path)).toStdString();
+        }
+        InitError(error);
+        QMessageBox::critical(nullptr, PACKAGE_NAME, QString::fromStdString(error.translated));
```

Comment:
re: https://github.com/bitcoin-core/gui/pull/602#discussion_r892764135

This wouldn't be intended, but I don't think this problem happens in practice, or at least I can't reproduce it. I can trigger the error with `settings.json` file:

```json
{
    "dbcache": [
    ]
}
```

In this case, InitError just logs an error without creating a message box because `app.createWindow` has not been called yet. After that the `QMessageBox::critical` call actually shows a message box.

There is a bunch of other in code the `src/qt/bitcoin.cpp` file following the pattern of calling `InitError` then `QMessageBox::critical`. It might be something that can be simplified in the future, but I think it is all correct and not buggy.

Reply:
ah ok, that was a bit misleading when I read it. Make sense, thanks 👌🏼